File: day1part1.js
const fs = require('node:fs');

const lines = fs.readFileSync('day1in.txt', 'utf8')

let a  = []
let b = []

lines.split("\n").forEach(function (line) {
    let nums = line.split(" ").filter(i => i !== '').map(i => parseInt(i))
    let first = nums[0]
    let second = nums[1]
    a.push(first)
    b.push(second)
})
a.sort()
b.sort()
let dist = 0
a.forEach(function (arg, i) {
    dist += Math.abs(b[i] - arg)
})
console.log(dist)

File: day1part1.rb
require 'byebug'

file = File.open "day1in.txt"

a = []
b = []
file.each_line do |l|
  erm1, erm2 = l.split(" ").filter { |it| it != " " }
  a << erm1.to_i
  b << erm2.to_i
end
a.sort!
b.sort!
dist = 0
a.each_with_index do |arg, i|
  dist += (arg - b[i]).abs
end
puts dist